### PR TITLE
Granularize Cartridge Reloading Part 1: Gotta keep em separated

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
```SUMMARY: Balance "Separates gunpowders by type"```

#### Purpose of change
A distinction needs to be made in terms of desperate back of the truck hand-loading, and hand-loading with proper components and tools. Many of these project steps and goals are outlined in the project of the same name.

#### Describe the solution
Pulls of gunpowder from bullets will be separated by type into 4 broad categories: shotgun/pistol, magnum pistol, rifle, and large rifle.  To make downloaded, inaccurate 'reloaded' ammunition, either these powders or 'mixed powder' can be used. 

Mixed powder is the above powders mixed and somewhat homogenized. This provides better justification for the 'reloaded' rounds' worse performance, due to their smaller safety margins and subsequent down-loading.

Proper powder, can, in a future PR, be used with other proper components, to make factory quality ammunition.

Artillery powder has been added too- single base(nitrocellulose) powder of similar composition, but different granule size. 

#### Describe alternatives you've considered
Discarding mixed powder.

### CODE REVIEW

**Comment by tenmillimaster** [comment]
First pass of powder overlaps made, but...

I need to identify archetypal powders for each type and then determine appropriate ratios,  and also where the overlaps lie.

Pistol/Shotgun - Accurate Solo 1000
Magnum pistol - Accurate 5744/ No. 9 (maybe not 5744, that has 300 win mag loads and might not be typical enough)
Rifle -  Accurate 2015
Large rifle - accurate 8700. .50bmg all the way down to .30-06.

File: data/json/recipes/ammo/grenades.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "charges": 1,
     "using": [ [ "ammo_40mm", 1 ] ],
     "tools": [ [ [ "press", -1 ] ] ],
-    "components": [ [ [ "gunpowder", 30 ] ], [ [ "combatnail", 25 ] ] ]
+    "components": [ [ [ "gunpowder", 30 ], [ "gunpowder_pistol", 30 ], [ "chem_black_powder", 30 ] ], [ [ "combatnail", 25 ] ] ]
```
Comment (Rail-Runner) [inline data/json/recipes/ammo/grenades.json]: Wouldn't blackpowder loads need to be a completely separate thing? They would create fouling that needs cleaning and smoke clouds, for instance.

Comment (tenmillimaster) [reply data/json/recipes/ammo/grenades.json]: Oh, that's true.

**Comment by tenmillimaster** [comment]
Archetypal powders and overlaps is looking a bit involved, so time to close this PR out:

The powders are split now, to shotguns, pistols, magnums, rifles, and overbore rifles. (Also artillery for people who want to shoot their vintage Bofors and what have you.)   Recipes for reloaded ammo have been adjusted, roughly to what I could find  overlaps for in  terms of powder (e.g. using shotgun powder for pistols, or magnum powder for intermediate rifles).

Makeshift reloading in the back of your truck has been retained.

You can now strain your pasta.

Next on the list will be:

- separating item groups for commercially sold reloading components and reloader's inventory components, adding more spawns as appropriate

- compiling powder tables and determining specific ratios for reloading. For example, reloading .223 with rifle powder would take 5 charges but with pistol would take 2, etc. This is looking fairly involved and I'm excited there are numbers to crunch and tables to make.

**Comment by Svankensen** [comment]
It is not venting. The gun system is genuinely hostile. What would be the
way to actually have this discussion?

**Comment by dommon** [comment]
oh how quaint, annuda guntard autiste, reverting those commits gets boring really, why is burgerland infested with them?

**Pull request merged**